### ISSUE TITLE
Bug in RandomUniform Bias Initializer

### ISSUE THREAD

**Issue by Kenivia** [plaidml/plaidml#417]
This is a weird bug,
if I do this:

```
model = Sequential()
 b = keras.initializers.RandomUniform(minval=-1, maxval=-1, seed=rseed)
 model.add(layers.Dense(16,
                           input_shape=(1,1000),
                           activation=keras.activations.elu,
                           use_bias=True,
                           bias_initializer=b,

                           ))
model.add(layers.Dense(10,
                           activation='linear'  ,
                           ))

```

It works just fine, but when I do this,
`b = keras.initializers.RandomUniform(minval=0, maxval=0, seed=rseed)`
it raises an error, strangely from the next layer:
```
Using plaidml.keras.backend backend.
INFO:plaidml:Opening device "opencl_amd_stoney.0"
Traceback (most recent call last):
  File "C:/Users/allen/Desktop/Big Items/Downloads/tidy_up_version.py", line 48, in <module>
    model = init_model()
  File "C:/Users/allen/Desktop/Big Items/Downloads/tidy_up_version.py", line 44, in init_model
    activation='linear'  ,
  File "C:\Users\allen\AppData\Local\Programs\Python\Python36\lib\site-packages\keras\engine\sequential.py", line 181, in add
    output_tensor = layer(self.outputs[0])
  File "C:\Users\allen\AppData\Local\Programs\Python\Python36\lib\site-packages\keras\engine\base_layer.py", line 458, in __call__
    output = self.call(inputs, **kwargs)
  File "C:\Users\allen\AppData\Local\Programs\Python\Python36\lib\site-packages\keras\layers\core.py", line 879, in call
    output = K.dot(inputs, self.kernel)
  File "C:\Users\allen\AppData\Local\Programs\Python\Python36\lib\site-packages\plaidml\tile.py", line 256, in function
    operation = cls(*args, **kwargs)
  File "C:\Users\allen\AppData\Local\Programs\Python\Python36\lib\site-packages\plaidml\op.py", line 1568, in __init__
    x.shape.dtype, y.shape.dtype))
ValueError: Invalid dtype in multiplication: x.dtype='51', y.dtype='50'
```

**Comment by dgkutnic** [comment]
Hi @Kenivia,
Thanks for reporting this bug. I get the same `ValueError` as you in _both_ cases of the RandomUniform initializer that you provided. From a bit of debug, I believe this error is occurring because you are passing in the same value for `minval` and `maxval` of the initializer, and our backend is interpreting that as an invalid range.
However, both of these cases work just fine using a TensorFlow backend. This leads me to believe that perhaps the TF backend is using an inclusive range (i.e. `[-1, -1]`) while we are using an exclusive range (i.e. `[-1, -1)`).
@mwyi can you look into this a bit more?

**Comment by dgkutnic** [comment]
Another note: if you are trying to set the bias to zero, you probably should be using `keras.initializers.Zeros()` instead of using `keras.initializers.RandomUniform()` with a zero range.

**Comment by WLeeeee** [comment]
This is a data type issue.

You have such issue because the bias values of your previous layer are sampled from the same value, which is -1. In this case, keras makes your bias (i.e., b) holds the integer type, which results in data type mismatch during matrix multiplication float32 v.s. int32 between layers.

You can dump and check the weights of your previous layer: 
print(model.layers[0].get_weights())

To work around, you only have to change -1 to -1.0.

**Comment by dgkutnic** [comment]
The issues in the RandomUniform bias initializer have been addressed in PR #434.
